test/relay_hub_test.js

```diff
@@ -634,4 +679,66 @@ contract("RelayHub", function (accounts) {
             assert.equal(expected_recipient_balance.toString(), relay_recipient_balance_after.toString())
         });
     })
+
+    it("should revert relayed call on an attempt to withdraw deposit during relayed transaction", async function () {
+
+        let withdrawDuringRelayedCall = await sr.withdrawDuringRelayedCall();
+        assert.equal(withdrawDuringRelayedCall, false);
+        await sr.setWithdrawDuringRelayedCall(true);
+        withdrawDuringRelayedCall = await sr.withdrawDuringRelayedCall();
+        assert.equal(withdrawDuringRelayedCall, true);
+
+        let digest = await getTransactionHash(from, to, transaction, transaction_fee, gas_price, gas_limit, relay_nonce, rhub.address, relayAccount);
+        let sig = await getTransactionSignature(web3, from, digest);
+
+        assert.equal(0, await rhub.canRelay(relayAccount, from, to, transaction, transaction_fee, gas_price, gas_limit, relay_nonce, sig));
+
+        let res = await rhub.relayCall(from, to, transaction, transaction_fee, gas_price, gas_limit, relay_nonce, sig, {
+            from: relayAccount,
+            gasPrice: gas_price,
+            gasLimit: gas_limit_any_value
+        });
+        relay_nonce++;
+        let PostRelayedFailed = 3;
+        assert.equal("TransactionRelayed", res.logs[0].event);
+        assert.equal(PostRelayedFailed, res.logs[0].args.status);
+
+        // returning state to previous one
+        await sr.setWithdrawDuringRelayedCall(false);
```

Comment:
should be cleared in try/finally, or in afterEach: it would be very confusing if this test fails, and then doesn't clean up after itself, and cause other tests to fail.

Reply:
Fixed